pybamm/expression_tree/array.py

```diff
@@ -57,6 +57,30 @@ def __init__(
             name, domain=domain, auxiliary_domains=auxiliary_domains, domains=domains
         )
 
+    @classmethod
+    def _from_json(cls, snippet: dict):
+        instance = cls.__new__(cls)
+
+        if isinstance(snippet["entries"], dict):
+            matrix = csr_matrix(
+                (
+                    snippet["entries"]["data"],
+                    snippet["entries"]["row_indices"],
+                    snippet["entries"]["column_pointers"],
+                ),
+                shape=snippet["entries"]["shape"],
+            )
+        else:
+            matrix = snippet["entries"]
+
+        instance.__init__(
+            matrix,
+            name=snippet["name"],
+            domains=snippet["domains"],
```

Comment:
question: do we need to serialise domains (sorry if I've asked this before, feels like we might have)? Since we are targeting already discretised models, perhaps we don't need domains? Or is this for plotting etc.

Reply:
We don't need the domains for running the model alone, but when plotting is required and variables/meshes have been saved, the variable domains are used to match to the correct mesh when deserialising.